### PR TITLE
[RFC] common/build-style/: enable running tests in parallel

### PR DESCRIPTION
**Pull request by Logarithmus**
What do you think? Are there any caveats?

### CODE REVIEW

**Review by q66 — APPROVED** [review]

**Review by Piraty — APPROVED** [review]

**Comment by Logarithmus** [comment]
@q66 @Piraty if you approve, then why no merging? I guess you have write permission for the repo.

**Comment by ericonr** [comment]
Should we add `disable_parallel_tests` to override this? Some test suites are explicitly broken when run in parallel (iirc cups, for example).

I assume they left their approval so others could review.

**Comment by q66** [comment]
yeah i guess we should add that

most suites go through fine with parallel builds (and often it's desirable because a lot of test suites compile things and running it singlethreaded with a large suite can get really slow)

**Comment by Logarithmus** [comment]
> Should we add `disable_parallel_tests` to override this? Some test suites are explicitly broken when run in parallel (iirc cups, for example).
> 
> I assume they left their approval so others could review.

I've added `disable_parallel_check` option and tested it for `mold`, it works as intended.

A little TODO:
- document the new option in `Manual.md`
- try running tests for `cups*` packages and see if it fail without `disable_parallel_check`.

**Comment by Chocimier** [comment]
Some build styles already run tests in parallel, please add disable_parallel_check there too.
It can be simpler with `if [ -z "$disable_parallel_check" ] local _check_jobs="${makejobs}"; make ... ${_check_jobs}`

**Comment by Logarithmus** [comment]
> Some build styles already run tests in parallel, please add disable_parallel_check there too.
> It can be simpler with `if [ -z "$disable_parallel_check" ] local _check_jobs="${makejobs}"; make ... ${_check_jobs}`

Do you have a list of build systems with parallel tests out of the box? I only know about `cmake+ninja` & probably `meson`.

**Comment by Chocimier** [comment]
It's just meson.

File: common/build-style/cmake.sh [hunk 1]
```diff
@@ -113,7 +113,11 @@ do_check() {
 
 	: ${make_check_target:=test}
 
-	${make_cmd} ${make_check_args} ${make_check_target}
+	if [ -z "$disable_parallel_check" ]; then
+		${make_cmd} ${makejobs} ${make_check_args} ${make_check_target}
+	else
+		${make_cmd} ${make_check_args} ${make_check_target}
```
Comment (sgn) [inline common/build-style/cmake.sh:119]: This is incorrect. `ninja` will run in parallel by default. I would rather change `makejobs` in `common/xbps-src/libexec/xbps-src-docheck.sh`

File: common/environment/setup/sourcepkg.sh [hunk 2]
```diff
@@ -3,7 +3,7 @@
 
 ## VARIABLES
 unset -v pkgname version revision short_desc homepage license maintainer
-unset -v archs distfiles checksum build_style build_helper nocross broken
+unset -v archs distfiles checksum build_style build_helper nocross disable_parallel_check broken
```
Comment (sgn) [inline common/environment/setup/sourcepkg.sh:6]: Please move to near `disable_parallel_build`

File: common/build-style/cmake.sh (hunk 1 above)
Comment (Logarithmus) [reply common/build-style/cmake.sh:119]: > This is incorrect. `ninja` will run in parallel by default. I would rather change `makejobs` in `common/xbps-src/libexec/xbps-src-docheck.sh`

Sorry, but I can't find `makejobs` in `common/xbps-src/libexec/xbps-src-docheck.sh`. Where should I define it?

File: common/environment/setup/sourcepkg.sh (hunk 2 above)
Comment (Logarithmus) [reply common/environment/setup/sourcepkg.sh:6]: Fixed

File: common/build-style/cmake.sh (hunk 1 above)
Comment (sgn) [reply common/build-style/cmake.sh:119]: It's exported from outside. You can change there, though.

Comment (Chocimier) [reply common/build-style/cmake.sh:119]: > Maybe we don't need `disable_parallel_check` altogether?

Successful multicore compilation with tests broken when running in parallel will likely happen.

> `disable_parallel_build` also implies `disable_parallel_check`, which isn't obvious

Documenting that should be enough. You could also export additional XBPS_MAX_MAKEJOBS or so and read it for check if disable_parallel_build is on if you prefer, just make sure to export that in right places.

**Pull request closed without merging**